libbpf-tools/tcppktlat.bpf.c

```diff
@@ -0,0 +1,99 @@
+// SPDX-License-Identifier: GPL-2.0
+// Copyright (c) 2023 Wenbo Zhang
+#include <vmlinux.h>
+#include <bpf/bpf_helpers.h>
+#include <bpf/bpf_core_read.h>
+#include <bpf/bpf_tracing.h>
+
+#include "compat.bpf.h"
+#include "tcppktlat.h"
+
+#define MAX_ENTRIES	10240
+#define AF_INET		2
+
+const volatile pid_t targ_pid = 0;
+const volatile pid_t targ_tgid = 0;
+const volatile __u16 targ_sport = 0;
+const volatile __u16 targ_dport = 0;
+const volatile __u64 targ_min_us = 0;
+
+struct {
+	__uint(type, BPF_MAP_TYPE_HASH);
+	__uint(max_entries, MAX_ENTRIES);
+	__type(key, u64);
+	__type(value, u64);
+} start SEC(".maps");
+
+SEC("tracepoint/tcp/tcp_probe")
+int tcp_probe_btf(struct trace_event_raw_tcp_probe *ctx)
+{
+	u64 sock_cookie, ts;
+
+	if (targ_sport && targ_sport != BPF_CORE_READ(ctx, sport))
+		return 0;
+	if (targ_dport && targ_dport != BPF_CORE_READ(ctx, dport))
+		return 0;
+	if (BPF_CORE_READ(ctx, data_len) == 0)
```

Comment:
Why do we need this filtering ?

`skb->data_len == 0` means the skb is linear.